lib/recipe.rb

```diff
@@ -0,0 +1,29 @@
+class Recipe
+  attr_reader :name, :ingredients_required
+
+  def initialize(name)
+    @name = name 
+    @ingredients_required = {}
+  end 
+
+  def add_ingredient(ingredient, quantity)
+    if @ingredients_required[ingredient].nil?
```

Comment:
Also could have set up the ingredients_required hash with a default value like you did in the pantry class, then you wouldn't need this if statement.